File: 2-Ad-Hoc/3478-Geiapan/3478.java
import java.util.Scanner;

public class Main {
    public static void main (String[] args) {
        int ncities, nfrontiers, qnt = 0, city1, city2;
        Scanner scan = new Scanner(System.in);
        ncities = scan.nextInt();
        nfrontiers = scan.nextInt();
        boolean cities[] = new boolean[ncities];
        for (int i = 0; i < ncities; ++i) {
            cities[i] = false;
        }
        for (int i = 0; i < nfrontiers; ++i) {
            city1 = scan.nextInt();
            city2 = scan.nextInt();
            if (!cities[city1-1] && !cities[city2-1]) {
                ++qnt;
            }
            cities[city1-1] = true;
            cities[city2-1] = true;
        }
        for (int i = 0; i < ncities; ++i) {
            if (!cities[i]) ++qnt;
        }
        System.out.println(qnt);   
    }
}

File: 2-Ad-Hoc/3478-Geiapan/3478.py
ncities, nfrontiers = input().split()
ncities, nfrontiers = int(ncities), int(nfrontiers)
cities = [False] * ncities
qnt = 0
for i in range (nfrontiers):
    c1, c2 = input().split()
    c1, c2 = int(c1), int(c2)
    if not cities[c1-1] and not cities[c2-1]:
        qnt += 1
    cities[c1-1], cities[c2-1] = True, True
print (qnt + cities.count(False))